modules/EVSE/EvseManager/EvseManager.cpp

```diff
@@ -309,6 +310,138 @@ void EvseManager::ready() {
+                hw_capabilities.max_phase_count_import == 3 ? std::make_optional(max_charge_power_per_phase)
+                                                            : std::nullopt,
+            };
+            ac_maximum_power.discharge_power.emplace(types::units::Power{
+                max_discharge_power_per_phase * hw_capabilities.max_phase_count_export,
+                max_discharge_power_per_phase,
+                hw_capabilities.max_phase_count_export >= 2 ? std::make_optional(max_discharge_power_per_phase)
+                                                            : std::nullopt,
+                hw_capabilities.max_phase_count_export == 3 ? std::make_optional(max_discharge_power_per_phase)
+                                                            : std::nullopt,
+            });
+            r_hlc[0]->call_update_ac_maximum_limits(ac_maximum_power);
+
+            types::iso15118::AcEvseMinimumPower ac_minimum_power;
+            const float min_charge_power_per_phase = config.ac_nominal_voltage * hw_capabilities.min_current_A_import;
+            const float min_discharge_power_per_phase =
+                config.ac_nominal_voltage * hw_capabilities.min_current_A_export;
+
+            ac_minimum_power.charge_power = {
+                min_charge_power_per_phase * hw_capabilities.max_phase_count_import,
+                min_charge_power_per_phase,
+                hw_capabilities.max_phase_count_import >= 2 ? std::make_optional(min_charge_power_per_phase)
+                                                            : std::nullopt,
+                hw_capabilities.max_phase_count_import == 3 ? std::make_optional(min_charge_power_per_phase)
+                                                            : std::nullopt,
+            };
+
+            ac_minimum_power.discharge_power.emplace(types::units::Power{
+                min_discharge_power_per_phase * hw_capabilities.max_phase_count_export,
+                min_discharge_power_per_phase,
+                hw_capabilities.max_phase_count_export >= 2 ? std::make_optional(min_discharge_power_per_phase)
+                                                            : std::nullopt,
+                hw_capabilities.max_phase_count_export == 3 ? std::make_optional(min_discharge_power_per_phase)
+                                                            : std::nullopt,
+            });
+            r_hlc[0]->call_update_ac_minimum_limits(ac_minimum_power);
+
+            std::vector<types::iso15118::Connector> ac_connectors{types::iso15118::Connector::SinglePhase};
+            if (get_hw_capabilities().max_phase_count_import == 3) {
+                ac_connectors.push_back(types::iso15118::Connector::ThreePhase);
+            }
+            r_hlc[0]->call_update_ac_parameters({50, static_cast<float>(config.ac_nominal_voltage), ac_connectors,
+                                                 std::nullopt, std::nullopt}); // TODO(sl): Getting nominal frequency
```

Comment:
This gets problematic if the bsp publishes its capabilities after this code is executed (e.g. when the BSP first has to wait for hardware to respond, like it is done [in the YetiDriver](https://github.com/EVerest/everest-core/blob/f7c4c8d38c3118d19b2da102eb59cdc0ac3fb463/modules/HardwareDrivers/EVSE/YetiDriver/board_support/evse_board_supportImpl.cpp#L143)).

Maybe put this code in a `bsp->subscribe_capabilities` callback?

Reply:
Fixed!